chains/evm/calls/bridge/bridge.go

```diff
@@ -0,0 +1,286 @@
+package bridge
+
+import (
+	"bytes"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/client"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/consts"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/contract"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/transactor"
+	"math/big"
+	"strings"
+
+	"github.com/ChainSafe/chainbridge-core/chains/evm/voter/proposal"
+	"github.com/ChainSafe/chainbridge-core/relayer/message"
+	"github.com/ChainSafe/chainbridge-core/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog/log"
+)
+
+type BridgeContract struct {
+	contract.Contract
+}
+
+func NewBridgeContract(
+	client client.ContractCallerDispatcherClient,
+	bridgeContractAddress common.Address,
+	transactor transactor.Transactor,
```

Comment:
Is there a usecase for read only contract instances?

Reply:
I would say there is, but this would mean an even bigger refactor - so I will make a discussion on this, and then we can see how could we implement read-only contracts.